src/controllers/account/account.ts

```diff
@@ -17,35 +31,99 @@ export const generateResetToken = async (): Promise<ResetToken> => {
   return { selector, verifier }
 }
 
-export const getEmailResetToken = async (input: ResetToken): Promise<string> => {
+// Abstract function to generate and store token and return for emailing
+export const generateAndStoreToken = async (ctx: BaseContext, longLiveToken: boolean): Promise<string> => {
+  const { email, role } = ctx.request.body
+  // Create activation token
+  const token = await generateAuthToken()
+
+  // Get verifier hash
+  const verifierHash = await getVerifierHash(token)
+
+  // Store json blob of request in dynamo db with key of activation token
+  await storeAuthToken(email, role, token.selector, verifierHash, longLiveToken)
+
+  // Send magic link via sendgrid helper service
+  const emailToken = await getEmailAuthToken(token)
+  return emailToken
+}
+
+export const getEmailAuthToken = async (input: AuthToken): Promise<string> => {
   return Buffer.concat([input.selector, input.verifier]).toString('base64')
 }
 
-export const getVerifierHash = async (input: ResetToken): Promise<Buffer> => {
+export const getVerifierHash = async (input: AuthToken): Promise<Buffer> => {
   const verifierHash = Buffer.alloc(sodium.crypto_generichash_BYTES)
   sodium.crypto_generichash(verifierHash, input.verifier)
   return verifierHash
 }
 
-// public static async storeResetToken (selector: Buffer, verifierHash: Buffer) {
-//   // TODO store selector in DB along with hash(verifier)
-// }
+export const storeAuthToken = async (email: string, role: string, selector: Buffer, verifierHash: Buffer, longterm: boolean) => {
+  // expiration will be UTC time since epoch of when this token expires
+  let expiration: number
```

Comment:
This block is complicated. How about this:
```
let expiration = Date.now()
if (longterm) {
  expiration += MILISECONDS_MONTH
} else {
  expiration += MILISECONDS_15_MINUTES
}
```

Reply:
done